File: agents/examples/typescript/cash_only.ts
// Reference participant (TypeScript): cash-only. Never trades. Ordinary external client.
import { clientFromEnv } from "../../../sdk/typescript/src/index.ts";

const STEP_MS = 6 * 3_600_000;

async function main() {
  const c = clientFromEnv();
  const info = (await c.describe()) as any;
  const duration: number = info.episode.duration_ms;
  for (;;) {
    const adv = await c.advance(Math.min(c.clockMs + STEP_MS, duration));
    if (adv.episode_ended) break;
  }
  const pf = await c.portfolio();
  const result = await c.finish();
  console.log("cash_only(ts) finished at", result.clock_ms, "equity", pf.valuation.model_equity_raw);
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: agents/examples/python/cash_only.py
"""Reference participant 1: cash-only. Never trades.

Verifies that abstention is a legitimate action and that cost accounting is stable.
Ordinary external client; no engine imports.
"""

from __future__ import annotations

import sys

from market_replay_client import client_from_env

STEP_MS = 6 * 3_600_000


def main(client=None) -> int:
    c = client or client_from_env()
    info = c.describe()
    duration = info["episode"]["duration_ms"]
    while True:
        adv = c.advance(min(c.clock_ms + STEP_MS, duration))
        if adv["episode_ended"]:
            break
    pf = c.portfolio()
    result = c.finish()
    print("cash_only finished at", result["clock_ms"], "equity", pf["valuation"]["model_equity_raw"])
    return 0


if __name__ == "__main__":
    sys.exit(main())
